### PR TITLE
Add some exceptions and handle it

### PR DESCRIPTION
**Pull request by boscohyun**
SSIA

image

image

image

### CODE REVIEW

File: src/main/main.ts [hunk 1]
```diff
@@ -673,10 +685,24 @@ async function initializeStandalone(): Promise<void> {
 
           break;
         } catch (error) {
-          console.error(
-            `Unexpected error occurred during download / extract snapshot. ${error}` +
-              `path: ${path}`
-          );
+          errorOccurred = true;
+          const errorMessage = `Unexpected error occurred during download / extract snapshot.\n${error}`;
```
Comment (ipdae) [inline src/main/main.ts:689]: `path` is deleted. is this correct change?

File: src/main/main.ts [hunk 2]
```diff
@@ -613,6 +624,7 @@ async function initializeStandalone(): Promise<void> {
   }
 
   initializeStandaloneCts = CancellationToken.create();
+  let errorOccurred = false;
```
Comment (ipdae) [inline src/main/main.ts:627]: there are no use of this variable.

File: src/utils.ts [hunk 3]
```diff
@@ -144,6 +151,10 @@ export async function cancellableExtract(
...
       `Unexpected error occurred during extracting ${targetDir} to ${outputDir}. ${error}`
     );
-    throw error;
+    throw new CancellableExtractFailedError(targetDir, outputDir);
   }
 }
+
+export function getType(target: any) {
+  return Object.prototype.toString.call(target).slice(8, -1);
+}
```
Comment (ipdae) [inline src/utils.ts:160]:
```suggestion
}

```

File: src/main/main.ts (hunk 1 above)
Comment (boscohyun) [reply src/main/main.ts:687]: Yes it is. The `path` used as argument of `snapshot.downloadMetadata()` and `snapshot.downloadSnapshot()`. And that used to make the `downloadPath` variable inside of these methods. Each methods throw new exceptions now(`DownloadSnapshotFailedError`, `DownloadSnapshotMetadataFailedError`) and these exceptions error messages contains the `downloadPath`.

**Review by riemannulus — APPROVED** [review]

**Review by area363 — APPROVED** [review]

**Pull request merged**